server/middleware/setupPdfRenderer.ts

```diff
@@ -0,0 +1,40 @@
+import { Request, Response, NextFunction } from 'express'
+import GotenbergClient, { PdfOptions } from '../data/gotenbergClient'
+import logger from '../../logger'
+
+/*
+ * This function accepts a Gotenberg client as its only argument.
+ * It returns a handler function to render a normal HTML view template to
+ * produce the raw HTML (including images, stylesheet etc). It then uses a
+ * callback function to pass rendered HTML view into the Gotenberg client
+ * to produce and return a PDF document.
+ */
+
+export default function setupPdfRenderer(client: GotenbergClient) {
```

Comment:
This function inspired heavily by the version in CVL 😁 (they are different though)